grpc/transport-auth.go

```diff
@@ -0,0 +1,70 @@
+package grpc
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/credentials"
+)
+
+// MultiNameAuthenticator does a thing
+type MultiNameAuthenticator struct {
+	names map[string]*tls.Config
+}
+
+// NewMultiNameTLS also does a thing
+func NewMultiNameTLS(names map[string]*tls.Config) credentials.TransportAuthenticator {
```

Comment:
Rather than taking this map, which is fairly low-level, what about taking a list of hostports, a root CA, and a client cert? Then MultiNameAuthenticator can construct the map itself and be confident in owning it exclusively.

This returns `credentials.TransportAuthenticator` but in the [current docs](https://godoc.org/google.golang.org/grpc/credentials#TransportCredentials) it seems to be called TransportCredentials. Do we need an additional library update?

Can we call this something shorter, like `boulder/grpc.ClientCredentials` ?